Move from labeling contents of /var/run to /run

LorbusChris opened the pull request:
This commit switches labelling from /var/run to /run.
The equivalence rule for /var/run and /run is rearranged to
better reflect the reality /var/run being a symlink to /run.

LorbusChris commented:
@rhatdan @wrabcak I've resurrected this. Right now, all the labeling in `/var/run`, which is a symlink to `/run`, makes the policy confusing. How can we move forward with this? What else would it need to deprecate `/var/run`?

rhatdan commented:
I have no problem with moving it. it was originally there because most distributions still used /var/run.  But about 8-10 years ago the defaults changed to /run at least for Fedora and eventually RHEL.   There was an equivalency rule that said label /run like it was in /var/run which made sure /run was labelled correctly.  Switching to /run and then adding an equivalency rule for /var/run would make the policy better match reality.

LorbusChris committed:
Label contents of /run directly

This commit switches labelling from /var/run to /run.
The equivalence rule for /var/run and /run is rearranged to
better reflect the reality of /var/run being a symlink to /run
in most distros now.

LorbusChris commented:
@rhatdan I've updated `config/file_contexts.subs_dist` with the equivalency rules. Please let me know if there is anything else needed here!

LorbusChris commented:
At a quick glance, I'm not sure where the CI syntax error is coming from. It wasn't there before and the PR does not touch the mentioned line?!

wrabcak commented:
Hi @LorbusChris , 
I need to fix the CI, it's issue in policy sources unrelated to this PR, then I merge this PR and will see what happen. It looks like a quite dangerous change.

wrabcak commented:
@LorbusChris , 
CI Fixed. 
But we need to change this also in selinux-policy-contrib repo. Could you create same PR also against selinux-policy-contrib repo? 

Thanks,
Lukas.

LorbusChris commented on policy/modules/contrib/apache.fc:
```diff
@@ -218,5 +218,6 @@ ifdef(`distro_debian', `
-/var/run/dirsrv/admin-serv.*	gen_context(system_u:object_r:httpd_var_run_t,s0)
+/run/dirsrv/admin-serv.*	gen_context(system_u:object_r:httpd_var_run_t,s0)
 /opt/dirsrv/var/run/dirsrv/dsgw/cookies(/.*)?       gen_context(system_u:object_r:httpd_var_run_t,s0)
+#/opt/dirsrv/run/dirsrv/dsgw/cookies(/.*)?       gen_context(system_u:object_r:httpd_var_run_t,s0)
```
NTS

LorbusChris commented on policy/modules/contrib/dbus.fc:
```diff
@@ -26,10 +26,11 @@ ifdef(`distro_gentoo',`
 
 ifdef(`distro_redhat',`
 /var/named/chroot/var/run/dbus(/.*)?	gen_context(system_u:object_r:system_dbusd_var_run_t,s0)
+#/var/named/chroot/run/dbus(/.*)?	gen_context(system_u:object_r:system_dbusd_var_run_t,s0)
```
NTS

LorbusChris reviewed (commented):
Rebased (sorry for resurrecting this...).

There are two paths that might make sense to add with this change, I've added the comments below as markers

LorbusChris commented:
@wrabcak I know this is going to be a pain to review, but please take a look when you get the chance

wrabcak commented:
@zpytela, Please take a look. 

Thanks,
Lukas.

LorbusChris commented:
I still think this would be a nice to have, but I don't have the time to look after this PR.
